quality-debt: setup-modules/agent-deploy.sh — PR #1253 review feedback (critical)

## Unactioned Review Feedback

**Source PR**: #1253
**File**: `setup-modules/agent-deploy.sh`
**Reviewers**: gemini
**Findings**: 2
**Max severity**: critical

---

### CRITICAL: gemini (gemini-code-assist[bot])
**File**: `setup-modules/agent-deploy.sh:1`
![critical](https://www.gstatic.com/codereviewagent/critical.svg)

The script is missing `set -euo pipefail` right after the shebang. This is required by the repository's shell script style guide (line 10) and is crucial for robust error handling and script safety.

```suggestion
#!/usr/bin/env bash
set -euo pipefail
```

<details>
<summary>References</summary>

1. All scripts must use `set -euo pipefail` to ensure robust error handling. <sup>([link](https://github.com/marcusquinn/aidevops/blob/main/.gemini/styleguide.md))</sup>
</details>

[View comment](https://github.com/marcusquinn/aidevops/pull/1253#discussion_r2799100564)
---

### MEDIUM: gemini (gemini-code-assist[bot])
**File**: `setup-modules/agent-deploy.sh:364`
![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)

The style guide (line 14) requires a reason comment for any `shellcheck` disable directives. This one is missing its justification.

```suggestion
	# shellcheck disable=SC2064 # The variable is expanded at trap creation time, which is safe in this context.
```

<details>
<summary>References</summary>

1. ShellCheck violations must not be disabled without a comment explaining the reason. <sup>([link](https://github.com/marcusquinn/aidevops/blob/main/.gemini/styleguide.md))</sup>
</details>

[View comment](https://github.com/marcusquinn/aidevops/pull/1253#discussion_r2799100686)
---

---
_Auto-generated by `quality-feedback-helper.sh scan-merged`. Review each finding and either fix the code or dismiss with a reason._

--- a/setup-modules/agent-deploy.sh
+++ b/setup-modules/agent-deploy.sh
@@ -374,7 +374,7 @@ install_beads_binary() {
 
 	local tmp_dir
 	tmp_dir=$(mktemp -d)
-	# shellcheck disable=SC2064
+	# shellcheck disable=SC2064  # Intentional: $tmp_dir must expand at trap definition time, not execution time
 	trap "rm -rf '$tmp_dir'" RETURN
 
 	if ! curl -fsSL "$download_url" -o "$tmp_dir/$tarball_name" 2>/dev/null; then

--- a/setup-modules/mcp-setup.sh
+++ b/setup-modules/mcp-setup.sh
@@ -85,7 +85,8 @@ install_mcp_packages() {
 }
 
 resolve_mcp_binary_path() {
-	local bin_name="$1"
+	local bin_name
+	bin_name="$1"
 	local resolved=""
 
 	# Check common locations in priority order
@@ -419,9 +420,10 @@ setup_browser_tools() {
 }
 
 add_opencode_plugin() {
-	local plugin_name="$1"
-	local plugin_spec="$2"
-	local opencode_config="$3"
+	local plugin_name plugin_spec opencode_config
+	plugin_name="$1"
+	plugin_spec="$2"
+	opencode_config="$3"
 
 	# Check if plugin array exists and if plugin is already configured
 	local has_plugin_array

